### PR TITLE
Migrate to have gen-payload apply imagestream updates

### PR DESCRIPTION
**Pull request by jupierce**
Requires https://github.com/openshift/doozer/pull/546

### CODE REVIEW

**Comment by jupierce** [comment]
This code touches several sensitive areas of our process. Merge must coincide with a low pressure window in the release schedule.

File: jobs/build/build-sync/build.groovy
Comment (vfreex) [inline jobs/build/build-sync/build.groovy:52]: This is already a named variable for the path of kubeconfig file:
```suggestion
    withEnv(["KUBECONFIG=${buildlib.ciKubeconfig}"]) {
```

**Review by vfreex — APPROVED** [review]
/lgtm
I added 1 suggestion to replace the hardcoded kubeconfig path with a buildlib variable.
/hold until the related Doozer PR lands.

**Comment by jupierce** [comment]
Replaced by: https://github.com/openshift/aos-cd-jobs/pull/2913

**Pull request closed without merging**
